pkg/ingestor/parser/csaf/parser_csaf_test.go

```diff
@@ -69,3 +72,70 @@ func Test_csafParser(t *testing.T) {
 		})
 	}
 }
+
+func Test_findPurl(t *testing.T) {
+	defaultTestTree := csaf.ProductBranch{
+		Name: "node1",
+		Branches: []csaf.ProductBranch{
+			{
+				Name: "node2",
+				Product: csaf.Product{
+					IdentificationHelper: map[string]string{
+						"purl": "test 2",
+					},
+				},
+			},
+		},
+	}
+
+	returnedPurl := defaultTestTree.Branches[0].Product.IdentificationHelper["purl"]
+	type args struct {
+		ctx         context.Context
+		tree        csaf.ProductBranch
+		product_ref string
+	}
+	tests := []struct {
+		name          string
+		args          args
+		stackOverflow bool
+		want          *string
+	}{
+		{
+			name: "default",
+			args: args{
+				ctx:         context.Background(),
+				tree:        defaultTestTree,
+				product_ref: "node2",
+			},
+			want: &returnedPurl,
+		},
+		{
+			name: "can stack overflow",
+			args: args{
+				ctx: context.Background(),
+				tree: csaf.ProductBranch{
+					Name: "node1",
+					Branches: []csaf.ProductBranch{
+						{
+							Name: "node2",
+						},
+					},
+				},
+				product_ref: "not equal to any tree nodes",
+			},
+			stackOverflow: true,
+			want:          nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.stackOverflow {
```

Comment:
same comment as other PR, can we make it just a regular test case and have a helper function create that test case.

Reply:
Fixed, did something similar to https://github.com/guacsec/guac/pull/1146#discussion_r1294811412